strawberry/field.py

```diff
@@ -196,10 +198,11 @@ def is_basic_field(self) -> bool:
 
     @property
     def arguments(self) -> List[StrawberryArgument]:
-        if not self.base_resolver:
-            return []
+        arguments = self.default_arguments[:]
+        if self.base_resolver and not self.ignore_resolver_arguments:
+            arguments.extend(self.base_resolver.arguments)
 
-        return self.base_resolver.arguments
+        return arguments
```

Comment:
I wonder if there's a more "elegant" solution to this than adding a `ignore_resolver_arguments` kwarg to `field`. Maybe move this from a property to something eagerly computed into a list on the field and then modify that list in the extension? We have the internal guarantee that field extensions are applied before the type is mapped to a gql-core type.

What do the others think?

Reply:
Yeah, I wrote this solution to avoid messing with the field API too much, and also to avoid changing the field extension code. I don't like it very much either.

Another solution could be adding an API to the field extension to resolve this. Something like this:

```python
@property
def arguments(self):
    arguments = self.base_resolver.arguments if self.base_resolver else []
    for extension in self.extensions:
        arguments = extension.resolve_arguments(self, arguments)
    return arguments
```

That `FieldExtension.resolve_arguments` would by default be implemented as:

```python
def resolve_arguments(self, field, arguments):
    return arguments
```

Meaning it would be a noop by default, but extensions could extend it or even remove arguments for advanced usage, like in this PR.

What do you think?

cc @patrick91 

obs. the `default_arguments` is also the same solution I used for the relay integration. Solving this in a more "elegant" way would also benefit it

Reply:
IMO we should leave `apply` as the only way to modify the attributes of a field. To make things a bit more ergonomic can we change the `arguments` property into a settable property? Something like this:

```python
    @property
    def arguments(self) -> List[StrawberryArgument]:
        if self._arguments:
            return self._arguments

        if not self.base_resolver:
            self._arguments = []
            return self._arguments

        self._arguments = self.base_resolver.arguments
        return self._arguments

    @arguments.setter
    def arguments(self, value: List[StrawberryArgument]):
        self._arguments = value
```

That way an extension can set the arguments in the apply method and it will be respected everywhere.

Reply:
I agree with @jkimbo that `apply` should remain the only place for field extensions to modify the field. This prevents a lot of unknowns as to execution order of `apply` & `resolve_arguments` removes the need for extensions to know about each other (e.g. one extension needs to know about the next extension's resolve_arguments result).

The approach with the setter seems to solve the problem well for that matter ☺️

Reply:
The setter should work indeed! I'll refactor the code to use that approach :)

Reply:
It's done!